packages/vercel-ai-provider/src/edgedb-provider.ts

```diff
@@ -0,0 +1,87 @@
+import { type Client, createClient } from "edgedb";
+import { getHTTPSCRAMAuth } from "edgedb/dist/httpScram.js";
+import { cryptoUtils } from "edgedb/dist/browserCrypto.js";
+import { getAuthenticatedFetch } from "edgedb/dist/utils.js";
+import type {
+  EmbeddingModelV1,
+  LanguageModelV1,
+  ProviderV1,
+} from "@ai-sdk/provider";
+import {
+  EdgeDBChatLanguageModel,
+  type EdgeDBLanguageModel,
+} from "./edgedb-chat-language-model";
+import type {
+  EdgeDBChatModelId,
+  EdgeDBChatSettings,
+} from "./edgedb-chat-settings";
+import { EdgeDBEmbeddingModel } from "./edgedb-embedding-model";
+import type {
+  EdgeDBEmbeddingModelId,
+  EdgeDBEmbeddingSettings,
+} from "./edgedb-embedding-settings";
+
+const httpSCRAMAuth = getHTTPSCRAMAuth(cryptoUtils);
+
+export interface EdgeDBProvider extends ProviderV1 {
+  (modelId: EdgeDBChatModelId | EdgeDBEmbeddingModelId): LanguageModelV1;
+
+  languageModel(
+    modelId: EdgeDBChatModelId,
+    settings?: EdgeDBChatSettings,
+  ): EdgeDBLanguageModel;
+
+  textEmbeddingModel: (
+    modelId: EdgeDBEmbeddingModelId,
+    settings?: EdgeDBEmbeddingSettings,
+  ) => EmbeddingModelV1<string>;
+}
+
+export async function createEdgeDB(client: Client): Promise<EdgeDBProvider> {
+  const connectConfig = await client.resolveConnectionParams();
+
+  const fetch = await getAuthenticatedFetch(
+    connectConfig,
+    httpSCRAMAuth,
+    "ext/ai/",
+  );
```

Comment:
Mostly leaving this as a note for the future:

We can avoid having to construct an authenticated fetch in every package by having the `Client` instance return one. The client has already had to construct an authenticated fetch client, and in this particular case, if you are in a Node environment, the one the driver constructs uses the native node crypto instead of browser crypto like this.

Maybe an API like:

```ts
const fetch = await client.getAuthenticatedFetch({
  path: "ext/ai/",
});
```